### DISCUSSION TITLE
risc0 → python ZKVM

### DISCUSSION THREAD

**Discussion by GuangguangLu** [risc0/risc0#713] [Ideas]
Hello everyone, risc0 is a zkvm implemented in rust. Is there a zkvm project implemented in python? It can be better combined with python programs.

**Comment by jan-ferdinand** [comment]
One that I know of is [stark-brainfuck](https://github.com/aszepieniec/stark-brainfuck). However, it is meant as an explainer of how zkVMs work, and I'm pretty sure it's too slow & memory hungry for anything beyond toy problems.

**Comment by SchmErik** [comment]
> Hello everyone, risc0 is a zkvm implemented in rust. Is there a zkvm project implemented in python? It can be better combined with python programs.

We have a binary called r0vm that can run guest code. You could do something like `subproccess` calls in python to this binary to get it to work with python
